src/main/resources/META-INF/resources/primefaces/tabview/tabview.js

```diff
@@ -459,8 +461,36 @@ PrimeFaces.widget.TabView = PrimeFaces.widget.DeferredWidget.extend({
         else {
             this.show(newPanel);
 
-            if(this.hasBehavior('tabChange') && !silent) {
-                this.fireTabChangeEvent(newPanel);
+            if (!silent) {
+                if (this.hasBehavior('tabChange')) {
+                    this.fireTabChangeEvent(newPanel);
+                }
+                else if (this.cfg.multiViewState) {
+                    var options = {
+                        source: this.id,
+                        partialSubmit: true,
+                        /*
+                        TODO: How to set partialSubmitFilter to only send and process activeIndex-hidden-field to the server?
+                        partialSubmitFilter: "@none",
+                        partialSubmitFilter: ":input:hidden",
+                        partialSubmitFilter: "@form:" + this.id + '_activeIndex',
+                        partialSubmitFilter: ":" + this.cfg.formId + ":" + this.id + '_activeIndex',
+                        partialSubmitFilter: "@form:" + this.id + '_activeIndex',
+                        partialSubmitFilter: this.id + '_activeIndex',
+
+                        process: "@none",
+                        process: this.id + '_activeIndex',
+                         */
+                        partialSubmitFilter: this.id + '_activeIndex',
+                        process: this.id,
+                        ignoreAutoUpdate: true,
+                        params: [
+                            {name: this.id + '_activeIndex', value: this.getActiveIndex()}
+                        ]
+                    };
+
+                    PrimeFaces.ajax.Request.handle(options);
+                }
```

Comment:
For dynamic="true" we have activeIndex on the server. Also when tabview includes tabChange-Ajax-Event.
This tries to send activeIndex-updates to the server for the other usecases. Is there a other (better) way to do this. (At least this way seems to work.)
And i have to remove the comment afterwards.

Reply:
To me this seems perfectly reasonable to do.